pkg/controller/gcp/compute/network.go

```diff
@@ -0,0 +1,174 @@
+	// Error strings.
+	errNewClient    = "cannot create new Compute Service"
+	errNotNetwork   = "managed resource is not a Network resource"
+	errNameNotGiven = "name for networkExternal resource is not provided"
+
+	errNetworkUpdateFailed = "update of Network resource has failed"
+	errNetworkCreateFailed = "creation of Network resource has failed"
+	errNetworkDeleteFailed = "deletion of Network resource has failed"
+)
+
+// NetworkController is the controller for Network CRD.
+type NetworkController struct{}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *NetworkController) SetupWithManager(mgr ctrl.Manager) error {
+	r := resource.NewManagedReconciler(mgr,
+		resource.ManagedKind(computev1alpha1.NetworkGroupVersionKind),
+		resource.WithExternalConnecter(&networkConnector{client: mgr.GetClient()}),
+		resource.WithManagedConnectionPublishers())
+
+	name := strings.ToLower(fmt.Sprintf("%s.%s", computev1alpha1.NetworkKindAPIVersion, computev1alpha1.Group))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.Network{}).
+		Complete(r)
+}
+
+type networkConnector struct {
+	client       client.Client
+	newServiceFn func(ctx context.Context, opts ...option.ClientOption) (*googlecompute.Service, error)
+}
+
+func (c *networkConnector) Connect(ctx context.Context, mg resource.Managed) (resource.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.Network)
+	if !ok {
+		return nil, errors.New(errNotNetwork)
+	}
+	// TODO(muvaf): we do not yet have a way for configure the Spec with defaults for statically provisioned resources
+	// such as this. Setting it directly here does not work since managed reconciler issues updates only to
+	// `status` subresource. We require name to be given until we have a pre-process hook like configurator in Claim
+	// reconciler
+	if cr.Spec.Name == "" {
+		return nil, errors.New(errNameNotGiven)
+	}
+
+	provider := &gcpapis.Provider{}
+	n := meta.NamespacedNameOf(cr.Spec.ProviderReference)
+	if err := c.client.Get(ctx, n, provider); err != nil {
+		return nil, errors.Wrapf(err, "cannot get provider %s", n)
+	}
+
+	gcpCreds, err := gcpclients.ProviderCredentials(c.client, provider, googlecompute.ComputeScope)
+	if err != nil {
+		return nil, err
+	}
+	if c.newServiceFn == nil {
+		c.newServiceFn = googlecompute.NewService
+	}
+	s, err := c.newServiceFn(ctx, option.WithCredentials(gcpCreds))
+	if err != nil {
+		return nil, errors.Wrap(err, errNewClient)
+	}
+	return &networkExternal{Service: s, projectID: provider.Spec.ProjectID}, nil
+}
+
+type networkExternal struct {
+	*googlecompute.Service
+	projectID string
+}
+
+func (c *networkExternal) Observe(ctx context.Context, mg resource.Managed) (resource.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.Network)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errNotNetwork)
+	}
```

Comment:
It just occurred to me that all observe, create, and delete methods should update the `Ready` condition by using `cr.SetConditions(runtimev1alpha1.Available())` (or `Creating()`, `Deleting()` etc as appropriate). Sorry this is not obvious or documented - see the AWS ReplicationGroup controller for example.

Reply:
No idea why I skipped those. Added.